### ISSUE TITLE
Unet 训练完后在移动端效果不佳

### ISSUE THREAD

**Issue by dalidifang** [PaddlePaddle/PaddleSeg#1664]
1.环境配置
PaddlePaddle 2.2.1 
PaddleSeg 版本:https://github.com/PaddlePaddle/PaddleSeg.git (20211226)
训练的图片尺寸为100X100左右大小，不是一样的。

2.训练配置unet.yml 
batch_size: 8
iters: 10000
train_dataset:
  type: Dataset
  dataset_root: /home/aistudio/work
  train_path: /home/aistudio/work/train.txt
  num_classes: 2
  transforms:
    - type: ResizeStepScaling
      min_scale_factor: 0.5
      max_scale_factor: 2.0
      scale_step_size: 0.25
    - type: RandomPaddingCrop
      crop_size: [512, 512]
    - type: RandomHorizontalFlip
    - type: RandomDistort
      brightness_range: 0.4
      contrast_range: 0.4
      saturation_range: 0.4
    - type: Normalize
  mode: train

val_dataset:
  type: Dataset
  dataset_root: /home/aistudio/work
  val_path: /home/aistudio/work/eval.txt
  transforms:
    - type: Normalize
  mode: val
  num_classes: 2

test_dataset:
  type: Dataset
  dataset_root: /home/aistudio/work
  val_path: /home/aistudio/work/test.txt
  transforms:
    - type: Normalize
  mode: test
  num_classes: 2

optimizer:
  type: sgd
  momentum: 0.9
  weight_decay: 4.0e-5

learning_rate:
  value: 0.01
  decay:
    type: poly
    power: 0.9
    end_lr: 0.0

loss:
  types:
    - type: CrossEntropyLoss
  coef: [1]

model:
  type: UNet
  num_classes: 2
  use_deconv: False
  pretrained: Null

3.python 预测
通过PaddleSeg-release-2.3/predict.py 进行预测，效果不错
4.导出
通过PaddleSeg-release-2.3/export.py 导出
然后使用paddle_lite_opt --model_dir=output/interface --valid_target=arm --optimize_out=paddleSeg-20211226进行转成Lite模型
5.安卓 预测
使用human_segmentation_demo 进行预测，INPUT_SHAPE_DEFAULT 修改为1,3,100,100。使用训练的样本数据进行预测，跟在python上预测的结果差别较大。

**Comment by juncaipeng** [comment]
你可以贴一下你两次预测的结果图片，手机上预测完全错误吗？

**Comment by dalidifang** [comment]
是这样的手机上，个别图片预测也比较好。就是有一些存在差异

1. 导出过程

> 

> python PaddleSeg-release-2.3/export.py --config unet.yml --model_path output/best_model/model.pdparams --save_dir output/interface

W1227 18:26:46.039201  9768 device_context.cc:447] Please NOTE: device: 0, GPU Compute Capability: 7.0, Driver API Version: 11.0, Runtime API Version: 10.1
W1227 18:26:46.043644  9768 device_context.cc:465] device: 0, cuDNN Version: 7.6.
2021-12-27 18:26:51 [INFO]      Loaded trained params of model successfully.
/opt/conda/envs/python35-paddle120-env/lib/python3.7/site-packages/paddle/fluid/layers/utils.py:77: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  return (isinstance(seq, collections.Sequence) and
2021-12-27 18:26:53 [INFO]      Model is saved in output/interface.

paddle_lite_opt --model_dir=output/interface --valid_target=arm --optimize_out=foot_seg_my_fp32
Loading topology data from output/interface/model.pdmodel
Loading params data from output/interface/model.pdiparams
1. Model is successfully loaded!
2. Model is optimized and saved into foot_seg_my_fp32.nb successfully

[23.zip](https://github.com/PaddlePaddle/PaddleSeg/files/7779807/23.zip)
![Screenshot_20211227_183217_com baidu paddle lite demo human_segmentation](https://user-images.githubusercontent.com/94677810/147464080-8a9794dc-dc35-46f6-92f9-29e3967646cb.jpg)

**Comment by juncaipeng** [comment]
如果是部分图片在台式机和手机上是预测相同、部分图片手机上预测不太好，这个应该不是使用错误，估计是预测框架的错误，比较难排查，需要你提供可以复现差异的模型、图片、代码，让我们来排查。
